packages/chip/src/Chip/Chip.tsx

```diff
@@ -0,0 +1,132 @@
+import React, { useMemo } from 'react';
+import PropTypes from 'prop-types';
+
+import { cx } from '@leafygreen-ui/emotion';
+import InlineDefinition from '@leafygreen-ui/inline-definition';
+import { useDarkMode } from '@leafygreen-ui/leafygreen-provider';
+
+import { DismissButton } from '../DismissButton';
+import { getTruncatedName } from '../utils/getTruncatedName';
+
+import {
+  chipInlineDefinitionClassName,
+  chipTextClassName,
+  chipTextDisabledStyles,
+  chipTextDismissSizeStyle,
+  chipTextSizeStyle,
+  chipTextStyles,
+  chipWrapperBaseDisabledStyles,
+  chipWrapperBaseStyle,
+  chipWrapperDisabledStyle,
+  chipWrapperSizeStyle,
+  chipWrapperThemeStyle,
+} from './Chip.styles';
+import { ChipProps, Size, TruncationLocation, Variant } from './Chip.types';
+
+export const Chip = React.forwardRef<HTMLSpanElement, ChipProps>(
+  (
+    {
+      chipTruncationLocation = TruncationLocation.none,
+      chipCharacterLimit = 15,
+      disabled = false,
+      size = Size.Default,
+      variant = Variant.Blue,
+      darkMode: darkModeProp,
+      label,
+      onDismiss,
+      popoverZIndex,
+      className,
+      dismissButtonAriaLabel,
+      ...rest
+    }: ChipProps,
+    forwardedRef,
+  ) => {
+    const { darkMode, theme } = useDarkMode(darkModeProp);
+
+    const isTruncated =
+      !!chipCharacterLimit &&
+      !!chipTruncationLocation &&
+      chipTruncationLocation !== 'none' &&
+      typeof label === 'string' &&
+      !!label &&
+      (label as string).length > chipCharacterLimit;
+
+    const truncatedName = useMemo(
```

Comment:
should only need to get this if `isTruncated` is true

Reply:
A few lines below (L94) there is a check for `isTruncated`. Would this still run with that check?